### PR TITLE
Adding name to some methods

### PR DESCRIPTION
**Pull request by antgonza**
~This is _not_ ready for review~

### CODE REVIEW

**Comment by josenavas** [comment]
This is ready for review

**Comment by antgonza** [comment]
Note that @josenavas and me pair programmed this.

**Comment by josenavas** [comment]
@wasade are you available for a review?

File: knimin/lib/data_access.py [hunk 1]
```diff
@@ -3477,6 +3477,8 @@ def extract_sample_plates(self, sample_plate_ids, email, robot, kit_lot,
             The name of the tool used for extraction
         notes : str, optional
             Notes added to the extracted plates
+        names : lilst of str, optional
```
Comment (wasade) [inline knimin/lib/data_access.py]: lilst -> list

File: knimin/lib/data_access.py [hunk 2]
```diff
@@ -3496,19 +3498,29 @@ def extract_sample_plates(self, sample_plate_ids, email, robot, kit_lot,
...
+                            email, name, created_on, sample_plate_id,
+                            extraction_robot_id, extraction_kit_lot_id,
+                            extraction_tool_id, notes)
+                         VALUES (%s, %s, now(), %s, %s, %s, %s, %s)
+                         RETURNING dna_plate_id"""
             dna_plates = []
-            for p_id in sample_plate_ids:
-                TRN.add(sql, [email, p_id, p_id, robot_id, kit_lot_id,
-                              tool_id, notes])
+            for p_id, name in zip_longest(sample_plate_ids, names,
```
Comment (wasade) [inline knimin/lib/data_access.py]: if `len(names) > len(sample_plate_ids)`, then a `None` will be used as the `p_id` placing a `null` in the database for the `sample_plate_id`. Is that correct? It looks like the following should be enforced:

```python
if names:
    if len(names) != len(sample_plate_ids):
        raise ValueError("...")
```

File: knimin/lib/data_access.py [hunk 3]
```diff
@@ -3460,7 +3460,7 @@ def get_sample_plate_list(self):
             return plates
 
     def extract_sample_plates(self, sample_plate_ids, email, robot, kit_lot,
-                              tool, notes=None):
+                              tool, notes=None, names=[]):
```
Comment (wasade) [inline knimin/lib/data_access.py]: it is dangerous to use a default like this, please change to `names=None`, test for it, and set the default within the method

Comment (ElDeveloper) [inline knimin/lib/data_access.py]: 👍

File: knimin/lib/tests/test_data_access.py [hunk 4]
```diff
@@ -1776,6 +1783,43 @@ def test_prepare_targeted_libraries(self):
...
+             'dna_plate_id': dna_plate_ids[1], 'primer_plate_id': 2,
+             'master_mix_lot': '14459', 'robot': 'ROBE',
+             'tm300_8_tool': '208484Z', 'tm50_8_tool': '108364Z',
+             'raw_concentration': None, 'mod_concentration': None,
+             'water_lot': 'RNBD9959'}]
+        for obs_id, exp in zip(obs_ids, exp):
+            obs = db.read_targeted_plate(obs_id)
+            self.assertTrue(before <= obs.pop('created_on') <= after)
+            for k in obs:
+                self.assertEqual(obs[k], exp[k])
```
Comment (ElDeveloper) [inline knimin/lib/tests/test_data_access.py]: Why is this per-key comparison needed if the following test is asserting the equality of the two objects as a whole?

File: knimin/lib/data_access.py [hunk 5]
```diff
@@ -4953,24 +4969,28 @@ def create_sequencing_run(self, pool_id, email, sequencer, reagent_type,
             The number of forward cycles used.
         rev_cycles : int
             The number of reverse cycles used.
+        name : str, optional
+            Name to add, if provided
```
Comment (ElDeveloper) [inline knimin/lib/data_access.py]: Can you note that if not provided, the default name will be as given by the pool i.e. what's happening in line 4982.

**Comment by antgonza** [comment]
@wasade and @ElDeveloper, could you take another look? Thanks!

File: knimin/lib/data_access.py [hunk 6]
```diff
@@ -3516,9 +3516,18 @@ def extract_sample_plates(self, sample_plate_ids, email, robot, kit_lot,
                          VALUES (%s, %s, now(), %s, %s, %s, %s, %s)
                          RETURNING dna_plate_id"""
             dna_plates = []
+
+            if names:
+                if len(names) != len(sample_plate_ids):
+                    raise ValueError(
```
Comment (ElDeveloper) [inline knimin/lib/data_access.py]: Can you add a note about this in the `Raises` section?

**Comment by ElDeveloper** [comment]
Looks good to me, @wasade can you merge this whenever you get a chance?

**Comment by ElDeveloper** [comment]
thanks @antgonza!

**Pull request merged**